aks/postgres/resources.tf

```diff
@@ -0,0 +1,246 @@
+locals {
+  azure_name                  = "${var.resource_group_name}-${var.app_environment}-postgres-${var.name}"
+  azure_private_endpoint_name = "${var.resource_group_name}-${var.app_environment}-postgres-${var.name}-pe"
+}
+
+resource "azurerm_postgresql_flexible_server" "main" {
+  count = var.use_azure ? 1 : 0
+
+  name                   = local.azure_name
+  location               = data.azurerm_resource_group.main.location
+  resource_group_name    = data.azurerm_resource_group.main.name
+  version                = var.server_version
+  administrator_login    = var.admin_username
+  administrator_password = var.admin_password
+  create_mode            = "Default"
+  storage_mb             = var.azure_storage_mb
+  sku_name               = var.azure_sku_name
+  delegated_subnet_id    = data.azurerm_subnet.main[0].id
+  private_dns_zone_id    = data.azurerm_private_dns_zone.main[0].id
+
+  dynamic "high_availability" {
+    for_each = var.azure_enable_high_availability ? [1] : []
+    content {
+      mode = "ZoneRedundant"
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      tags,
+      # Allow Azure to manage deployment zone. Ignore changes.
+      zone,
+      # Allow Azure to manage primary and standby server on fail-over. Ignore changes.
+      high_availability[0].standby_availability_zone,
+      # Required for import because of https://github.com/hashicorp/terraform-provider-azurerm/issues/15586
+      create_mode
+    ]
+  }
+}
+
+resource "azurerm_postgresql_flexible_server_configuration" "azure_extensions" {
+  count = var.use_azure ? 1 : 0
+
+  name      = "azure.extensions"
+  server_id = azurerm_postgresql_flexible_server.main[0].id
+  value     = "PG_BUFFERCACHE,PG_STAT_STATEMENTS,PGCRYPTO,UNACCENT"
```

Comment:
extensions should be passed to the module

Reply:
I've made it a variable, wasn't sure if this should be a default or not but I made it one. I'd expect most services want those extensions without having to worry about specifying them each time.